### PR TITLE
Fix: 'az redis update' operation for RDB/AOF enabled redis caches

### PR DESCRIPTION
**Pull request by MECHANDR**
"az redis update" fails on caches with RDB/AOF enabled, as the connecting string retrieved from GET doesn't have the key and is being passed in the PATCH request causing failure. Trimming the RDB/AOF connection string retrieved from GET will allow the cache PATCH to succeed.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@jsntcy  please help with the PR.

File: src/azure-cli/azure/cli/command_modules/redis/custom.py [hunk 1]
```diff
@@ -37,12 +37,18 @@ def cli_redis_update(cmd, instance, sku=None, vm_size=None):
...
     if instance.sku.name == 'Basic':
-        if 'maxmemory-reserved' in instance.redis_configuration:
-            instance.redis_configuration.pop('maxmemory-reserved')
-        if 'maxfragmentationmemory-reserved' in instance.redis_configuration:
-            instance.redis_configuration.pop('maxfragmentationmemory-reserved')
-        if 'maxmemory-delta' in instance.redis_configuration:
-            instance.redis_configuration.pop('maxmemory-delta')
+        memory_configs = ['maxmemory-reserved', 'maxfragmentationmemory-reserved', 'maxmemory-delta']
+        for memory_config in memory_configs:
+            if memory_config in instance.redis_configuration:
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/redis/custom.py]: Are there any tests that cover this change?

**Comment by jsntcy** [comment]
Sure

---
In reply to: [562803341](https://github.com/Azure/azure-cli/pull/11502#issuecomment-562803341) [](ancestors = 562803341)

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -117,6 +117,7 @@ Release History
 
 **Redis**
 
+* Fixing `az redis update` operation to work for caches with RDB/AOF enabled
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst]: - Is there any GitHub issue related to this PR?
If yes, please add issue link in PR description.
- Please also refine the PR title to <type>[optional scope]: <description> to make title more clear for others, you can reference to [https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)

Comment (MECHANDR) [reply src/azure-cli/HISTORY.rst]: > Is there any GitHub issue related to this PR?
> If yes, please add issue link in PR description.

There is no GitHub issue. Issue was found with support ticket from customer.
> Please also refine the PR title, for examples, "fix:Trimming RDB/AOF connection string in redis cache update operation" to make title more clear for others, you can reference to https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary

Updated.

File: src/azure-cli/azure/cli/command_modules/redis/custom.py (hunk 1 above)
Comment (MECHANDR) [reply src/azure-cli/azure/cli/command_modules/redis/custom.py]: No. Existing tests doesn't cover this. To test this scenario, there needs be an existing redis cache instance. Redis cache creation takes a while and as per the earlier guidelines, test wasn't allowed wait for longer duration.

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst]: Should not edit History in 2.0.77 section.
Please move it under *Release History* section: **Packaging** below  and  **Storage** above

Comment (MECHANDR) [reply src/azure-cli/HISTORY.rst]: Moved.

File: src/azure-cli/azure/cli/command_modules/redis/custom.py (hunk 1 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/redis/custom.py]: So how can we ensure your change is correct if no test covers this change?

---
In reply to: [357367878](https://github.com/Azure/azure-cli/pull/11502#discussion_r357367878) [](ancestors = 357367878)

Comment (MECHANDR) [reply src/azure-cli/azure/cli/command_modules/redis/custom.py]: As mentioned earlier, there needs to be an existing cache to add test coverage for this change. Please provide guidance on how to use sleep in tests until redis cache instance becomes available. Currently, I have tested this change locally.

**Pull request merged**